### PR TITLE
[BE]: ruff - enable PIE804

### PR DESCRIPTION
**Pull request by Skylion007**
Enables ruff PIE804 which kills some more unnecessary temporary dicts.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: test/torch_np/numpy_tests/core/test_numeric.py [hunk 1]
```diff
@@ -2991,14 +2991,14 @@ def test_broadcast_error_kwargs(self):
         # gh-13455
         arrs = [np.empty((5, 6, 7))]
         mit = np.broadcast(*arrs)
-        mit2 = np.broadcast(*arrs, **{})
+        mit2 = np.broadcast(*arrs, **{})  # noqa: PIE804
```
Comment (malfet) [inline test/torch_np/numpy_tests/core/test_numeric.py:2994]: Does it really need kwargs?
```suggestion
        mit2 = np.broadcast(*arrs)
```

File: test/torch_np/numpy_tests/core/test_numeric.py [hunk 2]
```diff
@@ -2991,14 +2991,14 @@ def test_broadcast_error_kwargs(self):
...
         assert_equal(mit.shape, mit2.shape)
         assert_equal(mit.ndim, mit2.ndim)
         assert_equal(mit.nd, mit2.nd)
         assert_equal(mit.numiter, mit2.numiter)
         assert_(mit.iters[0].base is mit2.iters[0].base)
 
-        assert_raises(ValueError, np.broadcast, 1, **{"x": 1})
+        assert_raises(ValueError, np.broadcast, 1, **{"x": 1})  # noqa: PIE804
```
Comment (malfet) [inline test/torch_np/numpy_tests/core/test_numeric.py:3001]: Just curious, why this wouldn't work?
```suggestion
        assert_raises(ValueError, np.broadcast, 1, x=1)
```

**Review by malfet — APPROVED** [review]

File: test/torch_np/numpy_tests/core/test_numeric.py (hunk 2 above)
Comment (Skylion007) [reply test/torch_np/numpy_tests/core/test_numeric.py:3001]: I just didn't want to touch tests here, I didn't know what was being used for torch compile etc and I did not want to reduce coverage.

File: test/torch_np/numpy_tests/core/test_numeric.py (hunk 1 above)
Comment (Skylion007) [reply test/torch_np/numpy_tests/core/test_numeric.py:2994]: if we remove it, then it's the same as mit 1 which seems like we are reducing coverage?

**Pull request closed without merging**